robot_smach_states/test/navigation/constraint_functions/test_arms_reach_constraints.py

```diff
@@ -0,0 +1,35 @@
+#! /usr/bin/env python
+import unittest
+
+from robot_skills.mockbot import Mockbot
+from robot_skills.util.kdl_conversions import frame_stamped
+from robot_smach_states.util.designators.core import Designator
+from robot_smach_states.util.designators.arm import ArmDesignator
+
+from robot_smach_states.navigation.constraint_functions.arms_reach_constraints import arms_reach_constraint
+
+
+class TestArmsReachConstraintFunction(unittest.TestCase):
+    def test_base(self):
+        robot = Mockbot()
+        arm = ArmDesignator(robot, {}, name="arm_designator")
+        frame = Designator(frame_stamped("/map", 0, 0, 0), name="frame_designator")
+
+        pc, oc = arms_reach_constraint(frame, arm)
+
+        self.assertIsNotNone(pc)
+        self.assertIsNotNone(oc)
+
+    def test_no_look(self):
+        robot = Mockbot()
+        arm = ArmDesignator(robot, {}, name="arm_designator")
+        frame = Designator(frame_stamped("/map", 0, 0, 0), name="frame_designator")
+
+        pc, oc = arms_reach_constraint(frame, arm, look=False)
+
+        self.assertIsNotNone(pc)
```

Comment:
What I'd  ke to see for all of these tests is to check the content of the `pc`, `oc` variables. 
The constraints are expressed at mathematical equations in a simple language, as a `str`. I'd like to be sure that the functions output the `str`s are correct. 

That holds for all of these tests.

Reply:
The tricky part is that we don't want to overly specify the format in which the string is constructed. For example "x>0" and "x > 0" are the same mathematical expressions but are different strings.

Reply:
I added as many checks as I found reasonable. But that is still not enough to verify the mathematical expressions are correct due to my previous point

Reply:
That is a good point. We could strip all whitespace of course but what matters is the underlying expression tree. 
For now, I'd be quite happy if we can get a few of such tests in. Just execute a test, record the result and check if it makes sense given the input. Then use that as the expected result. Then, we'll at least know when we break something.